migrations/v6-MIGRATION.md

```diff
@@ -54,22 +54,25 @@ StoreProduct has been made an interface, which `GoogleStoreProduct` and `AmazonS
 
 | Deprecated                                                                    | New                                                                                                 |
 |-------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------|
-| `purchaseProduct(Activity, StoreProduct, PurchaseCallback)`                   | `purchaseProductOption(Activity, StoreProduct, PurchaseOption, PurchaseCallback)`                   |
-| `purchaseProduct(Activity, StoreProduct, UpgradeInfo, ProductChangeCallback)` | `purchaseProductOption(Activity, StoreProduct, PurchaseOption, UpgradeInfo, ProductChangeCallback)` |
-| `purchasePackage(Activity, Package, PurchaseCallback)`                        | `purchasePackageOption(Activity, Package, PurchaseOption, PurchaseCallback)`                        |
-| `purchasePackage(Activity, Package, UpgradeInfo, ProductChangeCallback)`      | `purchasePackage(Activity, Package, PurchaseOption, UpgradeInfo, ProductChangeCallback)`            |
+| `purchaseProduct(Activity, StoreProduct, UpgradeInfo, ProductChangeCallback)` | `purchaseSubscriptionOption(Activity, StoreProduct, PurchaseOption, UpgradeInfo, ProductChangeCallback)` |
+| `purchasePackage(Activity, Package, UpgradeInfo, ProductChangeCallback)`      | `purchaseSubscriptionOption(Activity, StoreProduct, PurchaseOption, UpgradeInfo, ProductChangeCallback)` |
```

Comment:
These are deprecated because this is still a thing that is possible with one-time purchases, right?

Reply:
So basically upgrading from a different subscription is only available for subscriptions, so these methods should not be used for OTP. However, I thought some devs might want to upgrade to a different subscription while having us automatically choose the default offer/purchase option for them. That's why I thought these would be useful.

I updated the migration doc [here](https://github.com/RevenueCat/purchases-android/pull/713/commits/932a9bd8943c2b704262c6803d8141a3c9ab91d5)